schemas/v1.0/schema.yaml

```diff
@@ -1,5 +1,101 @@
-description: The description of Overlay v1.0.x documents
-
+description: The description of OpenAPI Initiative Overlay v1.0.0 documents
+  without schema validation, as defined by
+  https://spec.openapis.org/overlay/v1.0.0
 type: object
+$comment: https://spec.openapis.org/overlay/v1.0.0#overlay-object
+properties:
+  overlay:
+    description: The version number of the Overlay Specification
+    type: string
+    pattern: ^1\.0\.\d+(-.+)?$
+  info:
+    $ref: "#/$defs/info-object"
+  extends:
+    type: string
+    format: uri-reference
+  actions:
+    type: array
+    minItems: 1
+    uniqueItems: true
+    items:
+      $ref: "#/$defs/action-object"
+required:
+  - overlay
+  - info
+  - actions
+$ref: "#/$defs/specification-extensions"
+unevaluatedProperties: false
+$defs:
+  info-object:
+    $comment: https://spec.openapis.org/overlay/v1.0.0#info-object
+    description: Provides metadata about the Overlay description
+    type: object
+    properties:
+      title:
+        description: A human readable title of the Overlay Description
+        type: string
+      version:
+        description: The version identifier of the Overlay document (which is distinct
+          from the Overlay Specification version)
+        type: string
+    required:
+      - title
+      - version
+    $ref: "#/$defs/specification-extensions"
+    unevaluatedProperties: false
+  action-object:
+    $comment: https://spec.openapis.org/overlay/v1.0.0#action-object
+    description: This object represents one or more changes to be applied to the
+      target document at the location defined by the target JSONPath expression.
+    properties:
+      target:
+        description: A JSONPath expression selecting nodes in the target document
+        type: string
+        pattern: ^\$[.\[]?.*$
+      description:
+        description: A description of the action. [[CommonMark]] syntax MAY be used for
+          rich text representation.
+        type: string
+      update:
+        description: If the target selects an object node, the value of this field
+          MUST be an object with the properties and values to merge with the
+          node. If the target selects an array, the value of this field MUST
+          be an entry to append to the array.
+        type:
+          - string
+          - boolean
+          - object
+          - array
+          - number
+          - "null"
+      remove:
+        description: A boolean value that indicates that the target object is to be
+          removed from the the map or array it is contained in
+        type: boolean
+        default: false
+    allOf:
+      - description: The `update` field has no impact if the `remove` field of this
+          action object is true.
+        if:
+          properties:
+            remove:
+              const: true
+          required:
+            - remove
+        then:
+          not:
+            required:
+              - update
```

Comment:
This is stricter than the specification, which allows `update` together with `remove: true`.

I'd rather remove this, and also remove the test case `fail/actions-no-update-with-remove.yaml`.

Reply:
i asked for clarification on this in #71 

I think allowing this in the schema introduces an ambiguous implementation detail.. does the `remove` apply first, or the `update`?  Specifically because the spec indicates actions should be performed in order

Reply:
Let's wait for the outcome of #71 then with merging this PR.

Reply:
The schema validator we use thinks this `allOf` construct is invalid.